pkg/manifest/service.go

```diff
@@ -185,6 +185,19 @@ type ServiceResource struct {
 	Env  string
 }
 
+func (sr ServiceResource) GetConfigMapKey() string {
+	parts := strings.Split(sr.Env, "_")
+	key := parts[len(parts)-1]
+
+	for _, en := range AdditionalEnvNames {
+		if key == en {
+			return key
+		}
+	}
+
+	return DEFAULT_RESOURCE_ENV_NAME
```

Comment:
Sorry can you give some context about it?
why if doesn't match with anything we are returning `DEFAULT_RESOURCE_ENV_NAME`?

Reply:
sure thing!
when you create a resource, it will create a [ConfigMap](https://github.com/convox/convox/blob/0116fba7819c4284ba6103482adf83317e8eefa6/provider/k8s/template/resource/postgres.yml.tmpl#L12) with following keys: `URL`, `USER`, `PASS`, `HOST`, `PORT` and `NAME`.

The current behavior is to get the last part of the "linking name" and reference it in the environment variables pointing to the ConfigMap key. For example, setting `my-db:CUSTOM_DATABASE` in the service will create an environment variable called `CUSTOM_DATABASE` pointing to the `DATABASE` key in the ConfigMap. The database key doesn't exist in the CM and breaks the deployment.

To fix this, in case the user inserts a custom linking name, it will default to the URL key in the config map.